### ISSUE TITLE
Can't execute 'merge_base' when using Git.open.

### ISSUE THREAD

**Issue by madelaney** [ruby-git/ruby-git#397]
### Subject of the issue
I was trying to use the command `merge_base` on an existing repo setup, but ended up with the error message:
```ruby
irb(main):015:0> git.merge_base(git.current_branch, 'master')
NoMethodError: undefined method `merge_base' for #<Git::Base:0x00007fba8a83c510>
	from (irb):15
	from /usr/bin/irb:11:in `<main>'
```

### Your environment
* version of git and ruby-git
** git version 2.17.1 (Apple Git-112)
** ruby-git 1.4.0
* version of ruby
** ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]

### Steps to reproduce
* Clone an existing git repo
* Checkout non-master branch, and make changes.
* Using ruby-git
** Open dir using `Git.open('.')`
** Execute command `git.merge_base(git.current_branch, 'master')`

### Expected behaviour
I would expect to see a SHA1 returned that was the best match for a merge base between the current branch and master.

### Actual behaviour
NoMethodError error is thrown.

**Comment by madelaney** [comment]
When I look at the available methods this is the list that I get:
```ruby
irb(main):018:0> git.methods
=> [:grep, :index, :fetch, :push, :merge, :chdir, :with_index, :add, :dir, :reset, :config, :remove, :remotes, :lib, :repo, :checkout, :repo_size, :set_index, :set_working, :is_local_branch?, :is_remote_branch?, :is_branch?, :reset_hard, :clean, :describe, :revert, :commit, :commit_all, :checkout_file, :tags, :each_conflict, :pull, :add_remote, :set_remote_url, :remove_remote, :add_tag, :delete_tag, :archive, :repack, :gc, :apply, :apply_mail, :show, :with_temp_index, :checkout_index, :read_tree, :write_tree, :write_and_commit_tree, :update_ref, :ls_files, :with_working, :with_temp_working, :revparse, :ls_tree, :cat_file, :current_branch, :status, :tag, :log, :remote, :branch, :branches, :object, :commit_tree, :diff, :gblob, :gcommit, :gtree, :tap, :public_send, :instance_variables, :instance_variable_set, :instance_variable_defined?, :remove_instance_variable, :private_methods, :kind_of?, :is_a?, :instance_variable_get, :method, :public_method, :singleton_method, :instance_of?, :extend, :define_singleton_method, :to_enum, :enum_for, :<=>, :===, :=~, :!~, :eql?, :respond_to?, :freeze, :inspect, :object_id, :display, :send, :to_s, :nil?, :hash, :class, :singleton_class, :clone, :dup, :itself, :taint, :tainted?, :untaint, :untrust, :trust, :untrusted?, :methods, :protected_methods, :frozen?, :public_methods, :singleton_methods, :!, :==, :!=, :__send__, :equal?, :instance_eval, :instance_exec, :__id__]
```

**Comment by toaco** [comment]
I also encountered this problem.

**Comment by toaco** [comment]
it seems that Git 1.5 was released on 10 Aug 2018, but #370 was merge on 2 Oct 2018.
